### PR TITLE
Parallelize the unit tests

### LINKED ISSUE #6111: Improve unit test speed
This blog post provides a good background / introduction: https://threedots.tech/post/go-test-parallelism/

When I ran the following command
```shell
GOMAXPROCS=1 go test -parallel 128 -p 16 -json ./... | go run github.com/roblaszczak/vgt@latest
```

I got the following output:

Image

It shows some tests running for a very long time (the darker blue colors), but even of some of them were sped up there are others (like in the first group in top left corner) that are still relatively long.

We also almost never use `t.Parallel()` in our test (right now I only found one instance), even though many of our tests are starting servers and making network calls, so could benefit from parallelism. But adding `t.Parallel()` should be done with care - some tests are starting servers on the same port (instead of ephemeral port) and must be refactored before adding parallelism.

### PR DESCRIPTION
**Pull request by chahatsagarmain**
## Which problem is this PR solving?
- Part of #6111 

## Description of the changes
- Added t.parallel to tests where it made significant improvement 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: cmd/agent/app/reporter/grpc/builder_test.go [hunk 1]
```diff
@@ -300,10 +300,12 @@ func TestProxyClientTLS(t *testing.T) {
...
 		},
 	}
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
```
Comment (yurishkuro) [inline cmd/agent/app/reporter/grpc/builder_test.go:308]: what is this for?

**Comment by yurishkuro** [comment]
> Added t.parallel to tests where it made significant improvement

**Comment by yurishkuro** [comment]
please show evidence of improvement

**Comment by chahatsagarmain** [comment]
This is without `t.Parallel()`
Screenshot 2024-11-11 133952
After adding `t.Parallel()`
Screenshot 2024-11-11 134825

**Comment by chahatsagarmain** [comment]
Initially 
Screenshot 2024-11-11 140308
After adding `t.Parallel()` , the marked test are CollectorProxy test and ProxyClientTLS test which parallize 
Screenshot 2024-11-11 140806

**Comment by chahatsagarmain** [comment]
Initally 
Screenshot 2024-11-11 140308
After adding `t.Parallel()` , The marked ones are with `t.Parallel()`
parallel
TestProxyClientTLS is has the most time taking and in some runs it runs in an unordered manner without `t.Parallel()` .

**Comment by chahatsagarmain** [comment]
Initially 
Screenshot 2024-11-11 140308
Adding `t.Parallel()` to the 2 tests 
parallel
Trying to reduce the gap betweens tests

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**